### PR TITLE
From HackMD plan: manual, stress, web_platform, documentation, and some validation

### PR DESCRIPTION
**Pull request by kainino0x**
Part 1: everything except the guidelines and `api/validation/` and `api/operation/`, plus a few of the validation tests (the "weird" ones).

### CODE REVIEW

**Comment by kainino0x** [comment]
PTAL but feel free to ignore the test plans themselves (or not), but take a look at the file structure in case you have comments about it. Also be aware if I missed spelling out either `.unimplemented()` or the word "TODO:" anywhere (I don't think I did, but I want it to be ctrl-f-able - though at some point I need to make unimplemented tests actually findable in the standalone test plan viewer).

File: src/webgpu/api/validation/valid_to_use_with/lost_device/README.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Tests of behavior while the device is lost.
```
Comment (Kangz) [inline src/webgpu/api/validation/valid_to_use_with/lost_device/README.txt:1]: Also we could have the test of the error monad in valid_to_use_with, or do you think it should be in each of the function's tests?

File: src/webgpu/web_platform/canvas/context_creation.spec.ts [hunk 2]
```diff
@@ -5,19 +5,28 @@ import { makeTestGroup } from '../../../common/framework/test_group.js';
 
 export const g = makeTestGroup(Fixture);
 
-g.test('canvas_element_getContext_returns_GPUCanvasContext').fn(async t => {
-  if (typeof document === 'undefined') {
-    // Skip if there is no document (Workers, Node)
-    t.skip('DOM is not available to create canvas element');
-  }
+// TODO: Test all possible combinations of context creation attributes.
```
Comment (Kangz) [inline src/webgpu/web_platform/canvas/context_creation.spec.ts:8]: nit: remove this comment

**Review by Kangz — APPROVED** [review]
LGTM

File: src/webgpu/api/validation/valid_to_use_with/lost_device/README.txt (hunk 1 above)
Comment (kainino0x) [reply src/webgpu/api/validation/valid_to_use_with/lost_device/README.txt]: I added this, but decided later to remove it (it's more obvious to make it part of the entry point's tests, e.g. createView)

**Pull request merged**